broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java

```diff
@@ -741,7 +741,7 @@ public void run() {
                     log.error("registerBrokerAll Exception", e);
                 }
             }
-        }, 1000 * 10, 1000 * 30, TimeUnit.MILLISECONDS);
+        }, 1000 * 10, Math.max(10000, Math.min(brokerConfig.getRegisterNameServerPeriod(), 60000)), TimeUnit.MILLISECONDS);
```

Comment:
How do we prove this is a better value, why not other?

Reply:
Name servers will mark the broker as dead and remove routes to this broker if they do not receive registration heartbeat within 2T period. Empirically, brokers should send heartbeats every T moments at most to maintain their aliveness in case of occasional network failure, tolerating one-time heartbeat loss. 

As lite heartbeat mechanism has already been implemented, it's safe to heartbeat more frequently. This is why we allow interval of 10 seconds.